web/packages/teleterm/src/ui/services/workspacesService/workspacesService.ts

```diff
@@ -570,10 +570,17 @@ export class WorkspacesService extends ImmutableStore<WorkspacesState> {
     };
     for (let w in this.state.workspaces) {
       const workspace = this.state.workspaces[w];
+      // We don't persist 'doc.authorize_web_session' because we don't want to store
+      // a session token and id on disk.
+      // Moreover, the user would not be able to authorize a session at a later time anyway.
+      const documentsToPersist = (
+        workspace.previous?.documents || workspace.documents
+      ).filter(d => d.kind !== 'doc.authorize_web_session');
+
       stateToSave.workspaces[w] = {
         localClusterUri: workspace.localClusterUri,
         location: workspace.previous?.location || workspace.location,
```

Comment:
Ah, good point with the `location`. I The main reason I handled filtering out the documents before saving was this comment:
>On top of that, if we don't take any extra precautions, converting this modal into a tab means that data passed from the Web UI to Connect through the custom scheme URL would get saved on disk. Connect automatically saves the state of all open tabs on disk so that it's able to reopen tabs on the next launch.

> IIRC, the security model of Device Trust already accounts for this kind of data exfiltration. But it'd probably be a good idea to not persist this particular tab on disk in the first place.

The only kind of attack I can think of is authorizing a session with the user's confirmation. But you could only do it for a valid web request for a valid cluster, so I'm not sure if there's any benefit of that for an attacker. 
I agree that it's better to filter out the document when opening the app. However, maybe we should still clear the token and id from the document before saving to disk?

Reply:
I see, I will modify the fix I pushed.

Reply:
Restored previous code and added a test.